x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_event_type_signal.ts

```diff
@@ -25,5 +24,5 @@ export const buildEventTypeSignal = (doc: BaseSignalHit): object => {
  * @param doc The document which might be a signal or it might be a regular log
  */
 export const isEventTypeSignal = (doc: BaseSignalHit): boolean => {
-  return doc._source.signal?.rule?.id != null && typeof doc._source.signal?.rule?.id === 'string';
+  return doc._source?.signal?.rule?.id != null && typeof doc._source?.signal?.rule?.id === 'string';
```

Comment:
cc @dhurley14 @marshallmain

Reply:
`_source` is optional in the new type definitions. Hopefully, it will be fixed in https://github.com/elastic/elastic-client-generator/issues/197

Reply:
It is optional, _source can be false, and maybe we want to encourage use of the fields API instead anyway?